integration-test/src/main/java/com/sequenceiq/it/cloudbreak/dto/distrox/DistroXTestDtoBase.java

```diff
@@ -43,19 +43,17 @@ protected DistroXTestDtoBase(DistroXV1Request request, TestContext testContext)
         return this;
     }
 
-    public DistroXTestDtoBase<T> withEnvironment(Class<EnvironmentTestDto> environmentKey) {
+    public DistroXTestDtoBase<T> withEnvironment() {
         return withEnvironmentKey(EnvironmentTestDto.class.getSimpleName());
     }
 
     public DistroXTestDtoBase<T> withEnvironmentKey(String environmentKey) {
         EnvironmentTestDto env = getTestContext().get(environmentKey);
-        if (env == null) {
-            throw new IllegalArgumentException("Env is null with given key: " + environmentKey);
+        DistroXTestDtoBase<T> ret = this;
+        if (env != null && env.getResponse() != null) {
+            ret = withEnvironmentName(env.getResponse().getName());
         }
```

Comment:
I am missing the else branch where a illegalArgumentExcepton is thrown. What is the idea here?

Reply:
Not missing, just fetch the env name if it exists. Or we set later. After this the env can be null, I have moved the validation into the post action